### ISSUE TITLE
Mangled class names brake in IE8

### ISSUE THREAD

**Issue by kondi** [babel/babel#1369]
If you mangle the Babel output of a simple ES6 class using UglifyJS2, it won't work in IE8.

Let see the following ES6 code:

```javascript
class TestClass {
}
new TestClass();
```

The output will be:

```javascript
var TestClass = function TestClass() {
  _classCallCheck(this, TestClass);
};
new TestClass();
```

If you compress that using UglifyJS2:

```javascript
var TestClass = function X() {
  _classCallCheck(this, X);
};
new TestClass();
```

In IE8 the inner variable X and outer TestClass variables refer to two distinct function objects so the class call check fails. You can read more about it in the ``Example #3`` [of this nice article](https://kangax.github.io/nfe/#jscript-bugs).

There is a related ticket in UglifyJS2 repository (mishoo/UglifyJS2#248, especially [this comment](https://github.com/mishoo/UglifyJS2/issues/248#issuecomment-22758717)). In my opinion it is not easy to fix and can be circumvented if you change your code, so I don't expect a fix on the UglifyJS2 side for it soon (or at all...).

**Comment by monsanto** [comment]
Yes, those IE8 bugs are nasty. Perhaps we should have an optional transformer for converting named function expressions into regular functions?

**Comment by sebmck** [comment]
> Perhaps we should have an optional transformer for converting named function expressions into regular functions?

Seems like a better alternative than having the `es6.classes` transformer be aware of it. Could be called `jscript` or something.

**Comment by kondi** [comment]
Thank you for the quick feedbacks!

@monsanto If we would remove all the named function expressions, then the Function.prototype.name would not work correctly. Yes, I know, it doesn't work anyway in case of mangling, but for this specific bug we can have a less intrusive fix.

About the transformer, I completely agree, so I have created a new one called ``jscript`` as you proposed. This one can rewrite these unsafe expressions even if the user writes them, and retains the original function name.

I have filled a new PR #1382, then the other one can be closed I think.
